statsmodels/stats/rates.py

```diff
@@ -357,3 +964,516 @@ def tost_poisson_2indep(count1, exposure1, count2, exposure2, low, upp,
+
+    This reverses null and alternative hypothesis compared to equivalence
+    testing. The null hypothesis is that the effect, ratio (or diff), is in
+    an interval that specifies a range of irrelevant or unimportant
+    differences between the two samples.
+
+    The Null and alternative hypothesis comparing the ratio of rates are
+
+    - H0: low < g1 / g2 < upp
+    - H1: g1 / g2 <= low or upp <= g1 / g2
+
+    where g1 and g2 are the Poisson rates.
+
+
+    Notes
+    -----
+    This is implemented as two one-sided tests at the minimum effect boundaries
+    (low, upp) with (nominal) size alpha / 2 each.
+    The size of the test is the sum of the two one-tailed tests, which
+    corresponds to an equal-tailed two-sided test.
+    If low and upp are equal, then the result is the same as the standard
+    two-sided test.
+
+    The p-value is computed as `2 * min(pvalue_low, pvalue_upp)` in analogy to
+    two-sided equal-tail tests.
+
+    In large samples the nominal size of the test will be below alpha.
+
+    References
+    ----------
+    Hodges, J. L., Jr., and E. L. Lehmann. 1954. Testing the Approximate
+    Validity of Statistical Hypotheses. Journal of the Royal Statistical
+    Society, Series B (Methodological) 16: 261–68.
+
+    Kim, Jae H., and Andrew P. Robinson. 2019. “Interval-Based Hypothesis
+    Testing and Its Applications to Economics and Finance.”
+    Econometrics 7 (2): 21. https://doi.org/10.3390/econometrics7020021.
+
+    """
+    tt1 = test_poisson_2indep(count1, exposure1, count2, exposure2,
+                              ratio_null=low, method=method, compare=compare,
+                              alternative='smaller')
+    tt2 = test_poisson_2indep(count1, exposure1, count2, exposure2,
+                              ratio_null=upp, method=method,
+                              alternative='larger')
+
+    # idx_min = 0 if tt1.pvalue < tt2.pvalue else 1
+    idx_min = np.asarray(tt1.pvalue < tt2.pvalue, int)
+    pvalue = 2 * np.minimum(tt1.pvalue, tt2.pvalue)
+    np.column_stack([tt1.statistic, tt2.statistic])[idx_min]
+    res = HolderTuple(statistic=stats,
+                      pvalue=pvalue,
+                      method=method,
+                      results_larger=tt1,
+                      results_smaller=tt2,
+                      title="Equivalence test for 2 independent Poisson rates"
+                      )
+
+    return res
+
+
+def confint_poisson_2indep(count1, exposure1, count2, exposure2,
+                           method='score', compare='ratio', alpha=0.05,
+                           method_mover="score",
+                           ):
+    """Confidence interval for ratio of 2 indep poisson rates
+    """
+
+    # shortcut names
+    y1, n1, y2, n2 = count1, exposure1, count2, exposure2
+    rate1, rate2 = y1 / n1, y2 / n2
+    alpha = alpha / 2  # two-sided only
+
+    if compare == "ratio":
+
+        if method == "wald":
```

Comment:
no unit tests, consistency with confint ?
this looks like wald-log, and similar score-log below, and wald-logcc